core/domain/question_jobs_one_off.py

```diff
@@ -142,6 +143,67 @@ def reduce(key, values):
             yield (key, values)
 
 
+class DeleteInvalidQuestionModelsOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """This job is used to delete invalid question models and its linked models.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [question_models.QuestionModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        snapshot_model_ids = [
+            '%s-%d' % (item.id, version) for version in python_utils.RANGE(
+                1, item.version + 1)]
+        commit_log_model_ids = [
+            'question-%s-%s'
+            % (item.id, version) for version in python_utils.RANGE(
+                1, item.version + 1)]
+
+        # In the Audit job, the version 1 model for the 3 models below were
+        # absent. So, if they are absent, delete all versions and the linked
+        # QuestionModel. (This issue does not exist anymore)
```

Comment:
Do we need to update linked QuestionSkillLinkModels as well? (And are there any other models pertaining to questions that might be affected?)